openlibrary/i18n/__init__.py

```diff
@@ -101,6 +103,34 @@ def load_translations(lang):
     if os.path.exists(mo_path):
         return Translations(open(mo_path, 'rb'))
 
+
+def get_ol_locale():
+    """
+    Gets the locale from the cookie and -if not found- sets it to the
+    highest priority language of the Accept-Language header, that's also part
+    of the Open Library supported languages.
+    """
+    locale_cookie = web.cookies().get('i18n_code')
+    if locale_cookie:
+        ol_locale = locale_cookie
+    else:
+        browser_accepted_langs = web.ctx.env.get('HTTP_ACCEPT_LANGUAGE')
+        if browser_accepted_langs:
+            # Parse the accepted language header string into a list
+            languages = browser_accepted_langs.split(',')
+            # Remove the quality-value weights and keep only the locales
+            accepted_langs = [lang.partition(';')[0] for lang in languages]
+            # The default locale is 'en' unless we support a browser-suggested language
+            ol_locale = 'en'
+            for lang in accepted_langs:
+                if lang in OL_SUPPORTED_LANGAGES:
+                    ol_locale = lang
+                    break
```

Comment:
I believe this is already done be `web.ctx.get('lang', 'en')`; it should be checking `HTTP_ACCEPT_LANGUAGE` as well as the url parameter e.g. `?lang=fr` we use for testing.


Oh my... Now that I look at it looks like this is _all_ already done! The cookie is called `HTTP_LANG`. So all we have to do is set it, and everything should work? https://github.com/internetarchive/infogami/blob/03b7f1c38bae80c4930443f46a32b769cdd5fe57/infogami/utils/i18n.py#L132 Ahhh, sorry for the wild goose chaise :(

Reply:
Done. Using the `i18n_loadhook` now.